src/lib/rest/StringFilter.cpp

```diff
@@ -920,28 +926,20 @@ bool StringFilter::parse(const char* q, std::string* errorStringP)
       return false;
     }
 
-    StringFilterItem item;
+    StringFilterItem* item = new StringFilterItem();
 
-    if (item.parse(s, errorStringP) == true)
+    if (item->parse(s, errorStringP) == true)
     {
       filters.push_back(item);
     }
     else
     {
+      free(item);
```

Comment:
Better to use `delete'.
With `free`, the destructor is not called, and we've seen some problems in the past mixing C/C++ in malloc/new/free/delete situations. Not that I can recall any details  ...